kv/dist_sender.go

```diff
@@ -184,6 +186,10 @@ func NewDistSender(ctx *DistSenderContext, gossip *gossip.Gossip) *DistSender {
 	if ctx.RangeLookupMaxRanges <= 0 {
 		ds.rangeLookupMaxRanges = defaultRangeLookupMaxRanges
 	}
+	ds.tracer = ctx.Tracer
+	if ds.tracer == nil {
+		ds.tracer = util.NewTracer(nil, 2)
```

Comment:
It seems strange to create a Tracer here (same in store.go). Since the tracer values tolerate nil methods why not just leave it as nil unless a value was passed in?